mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -116,8 +116,7 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
 
     val distinctSplits = splits.distinct
     if (splits.length != distinctSplits.length) {
-      log.warn(s"Some quantiles were identical. Bucketing to ${distinctSplits.length - 1}" +
-        s" buckets as a result.")
+      log.warn(s"Some quantiles were identical. Will return fewer buckets than requested!")
```

Comment:
when the splits has NaN, we have to know the number of non-NaN value in the splits to get the exact number of returned buckets, considering splits :{ -Inf, -1, 1, Inf, NaN, NaN}, the old msg will actually get us a wrong result